x-pack/plugins/screenshotting/server/screenshots/wait_for_render.ts

```diff
@@ -30,7 +30,24 @@ export const waitForRenderComplete = async (
 
         function waitForRender(visualization: Element) {
           return new Promise<void>((resolve) => {
-            visualization.addEventListener('renderComplete', () => resolve());
+            const observer = new MutationObserver((mutationList) => {
+              for (const mutation of mutationList) {
+                if (mutation.type === 'attributes') {
+                  if (
+                    mutation.attributeName === 'data-render-complete' &&
+                    visualization.getAttribute('data-render-complete') === 'true'
+                  ) {
+                    observer.disconnect();
+                    resolve();
+                  }
+                }
+              }
+            });
+            observer.observe(visualization, { attributes: true });
```

Comment:
I didn't run this locally, but I suppose we are leveraging mutation observers because the `renderComplete` event no longer gets dispatched, I see that it ideally would have been dispatched here https://github.com/elastic/kibana/blob/main/src/plugins/visualizations/public/embeddable/visualize_embeddable.tsx#L392, is this intentional?

Reply:
Yes, in the new embeddable framework everything goes through React, so it's much more cumbersome to get the kind of DOM access we need to emit events from the correct element. We're trying to move away from event emitters in new code.

Reply:
My concern is that previously the implementation detail for when a render completes was contained, and any interested consumer of visualizations need only care if a `renderComplete` event was emitted on any said visualization element. I think that a replacement util for signalling `renderComplete` in the new code should be provided, as it rather seems that the implementation detail of determining if a render has completed is now being shifted on to the consumer.